### PR TITLE
nvme: add missing endianness annotations

### PR DESCRIPTION
**Pull request by ganaidu707**
Signed-off-by: Gollu Appalanaidu <anaidu.gollu@samsung.com>

### CODE REVIEW

**Comment by keithbusch** [comment]
You have them annotated correctly now, but didn't add any of the missing le*_to_cpu() in the accessors. I wonder if we can get incorporate something like the kernel's "sparse" make analysis to find these kinds of incorrect usages...

File: linux/nvme.h [hunk 1]
```diff
@@ -1458,21 +1458,21 @@ struct nvme_error_log_page {
 struct nvme_firmware_log_page {
 	__u8	afi;
 	__u8	resv[7];
-	__u64	frs[7];
+	__le64	frs[7];
```
Comment (keithbusch) [inline linux/nvme.h:1461]: this type may have incorrectly been carried over from tools implementing pre-releases of the spec. The field is an ASCII array, so it should be a __u8[8].

Comment (ganaidu707) [reply linux/nvme.h:1461]: yeah, I have fixed and added a PR https://github.com/linux-nvme/nvme-cli/pull/1088
Please have a look!
Thanks

**Comment by ganaidu707** [comment]
> You have them annotated correctly now, but didn't add any of the missing le*_to_cpu() in the accessors. I wonder if we can incorporate something like the kernel's "sparse" make analysis to find these kinds of incorrect usages...

Added missing le*_to_cpu() in the accessors.
Thanks!

**Pull request merged**